hudi-utilities/src/test/java/org/apache/hudi/utilities/testutils/UtilitiesTestBase.java

```diff
@@ -364,5 +385,32 @@ public static String toJsonString(HoodieRecord hr) {
     public static String[] jsonifyRecords(List<HoodieRecord> records) {
       return records.stream().map(Helpers::toJsonString).toArray(String[]::new);
     }
+
+    public static void addAvroRecord(
+            VectorizedRowBatch batch,
+            GenericRecord record,
+            TypeDescription orcSchema,
+            int orcBatchSize,
+            Writer writer
+    ) throws IOException {
+      for (int c = 0; c < batch.numCols; c++) {
+        ColumnVector colVector = batch.cols[c];
+        final String thisField = orcSchema.getFieldNames().get(c);
+        final TypeDescription type = orcSchema.getChildren().get(c);
+
+        Object fieldValue = record.get(thisField);
+        Schema.Field avroField = record.getSchema().getField(thisField);
+        AvroOrcUtils.addToVector(type, colVector, avroField.schema(), fieldValue, batch.size);
+      }
+
+      batch.size++;
+
+      if (batch.size % orcBatchSize == 0 || batch.size == batch.getMaxSize()) {
```

Comment:
can you help me understand what this code block is doing? I see this method is called for one batch of records. lets say there are 100 records in one batch. if I am not wrong, batch.size at the end of adding 100 records should be 100. and so batch.size % orcBatchSize will be equal to 0 only after adding all records. 
If thats the case, shouldn't we move this block outside of this method. 
or am I missing something.

Reply:
Sure thing, changed :)